### ISSUE TITLE
Slider congestion reserve neighbourhood battery keeps changing back to 20%

### ISSUE THREAD

**Issue by Sanne512** [quintel/etmoses#1119]
In the following LES [https://moses.energytransitionmodel.com/testing_grounds/471/edit](url) the slider for the congestion reserve of the neighbourhood battery changes back to 20%. I'm not sure when or why this happens exactly, but when I run into it again I will update this issue. 

@ChaelKruip who do I need to assign to fix this?

**Comment by Sanne512** [comment]
Update: I changed the output capacity and the volume of the Neighbourhood battery at Buurtbatterij (1) , clicked the blue save button and clicked the green save all and view les button. When I viewed the LES the slider had changed back to 20% instead of the 33% I set it to.

**Comment by antw** [comment]
> @ChaelKruip who do I need to assign to fix this?

Don't worry too much about that; we can assign to an appropriate person. :smiley:

I can't actually reproduce this myself, but I have noticed that <select> boxes are always showing the default (first) option instead of the real selected value.

**Comment by Sanne512** [comment]
Ok, this keeps happening everytime I work with the neighbourhood battery. As a work-around I save and load the LES one time to ensure the new numbers for capacity and volume are being changed. Then I reopen "edit the LES" to adjust the slider to 33% congestion again. This works, but it takes a lot of time to save and load the LES two times every time I change something.

**Comment by Sanne512** [comment]
@JeroenvdLogt and I discovered that the same thing happens to the profiles you change for the different technologies and profiles you add. When I change the profile of the space heating to tussenwoning_midden it keeps changing back to appartement_hoog (in the same LES) and for residential PV panel the profile keeps changing back from solar_pv_amsterdam to solar_pv_amsterdam. Jeroen has the same problem with profiles he created and added himself. 

Could you please look into this again if you can reproduce it? Because these errors make it really hard for us to use ETMoses at the moment.

**Comment by antw** [comment]
> Could you please look into this again if you can reproduce it?

I know this isn't very helpful, but I'm fairly sure the congestion percentage and profile settings *are* being saved. It's just that when you visit the "Edit LES" page it's showing you the *default* options instead of what is really selected. I'm looking into it now.

**Comment by Sanne512** [comment]
> I'm fairly sure the congestion percentage and profile settings are being saved

Not all the time. I tried to reproduce it myself in the LES I mentioned in the first post (buurtbatterij - overzicht) and couldn't do so. The slider was showing the default setting, but it calculated with the 33% I changed it to. 

However I reproduced the error it in the following LES: https://moses.energytransitionmodel.com/testing_grounds/476 by taking these steps: 

1. Opened the LES and saw in the electricity load graph that the slider was indeed set to 33%, because there was no congestion at the MSR. 
2. Clicked edit LES. I saw that the congestion slider of the neighbourhood battery showed that it was set to 20%. 
3. Without changing anything I clicked the blue save button. 
4. Clicked the green save all and view LES button. Looking at the electricity load graph now I know that it calculating with a congestion of lower than 33% because there is congestion at the MSR. 

Does this help?

**Comment by JeroenvdLogt** [comment]
It indeed looks like the settings are saved but not displayed correctly in the technology table (at least for the profiles, I don't know for the neighborhood battery). Strange thing is that it is correcly displayed in the overall dashboard.

https://moses.energytransitionmodel.com/testing_grounds/490 - eerste trial route kaar hof van twente → MV customer Amt Delden

![bug wijziging profielen1](https://cloud.githubusercontent.com/assets/7781037/16305936/3f4ded6c-395c-11e6-8ed5-8dc5625ab65c.JPG)
![bug wijziging profielen2](https://cloud.githubusercontent.com/assets/7781037/16305937/3f70be28-395c-11e6-869f-8032b1b4b113.JPG)

**Comment by antw** [comment]
I have fixed the technology profile drop-down showing the wrong option.

However, I can't reproduce any situation in which the congestion battery percentage is incorrect. I set it to 20% and the value was correctly saved (and refreshing the "Edit LES" page showed the correct value). I set it to 80% and, likewise, the value was saved and the slider showed the correct value.

> Opened the LES and saw in the electricity load graph that the slider was indeed set to 33%, because there was no congestion at the MSR.

Changing the congestion percentage should have no effect on the behaviour of the battery in times of network congestion. The slider controls what percentage of the battery is reserved for responding to periods of excess production or consumption in the LES.

If you set it to 40% it means the battery may not discharge below 20% of it's volume – or charge above 80% – unless doing so alleviates network congestion (by removing it entirely, or at least reducing the severity of the event). At 40% reserved, it means that 60% of the battery volume may be charged and discharged like any other P2P battery.

Basically: the slider controls the battery in times where there is no network congestion. When there *is* congestion, the battery will behave the same way regardless of what value you choose.

Here's an example from your LES. I set the reserve percentage to 20% and 80% respectively:

| 20% reserved | 80% reserved |
| --- | --- |
| ![screen shot 2016-06-23 at 15 31 15](https://cloud.githubusercontent.com/assets/4383/16308193/e9a34efc-395b-11e6-80a7-d779ae93278d.png) | ![screen shot 2016-06-23 at 15 31 20](https://cloud.githubusercontent.com/assets/4383/16308195/ef5d80b0-395b-11e6-8979-09fab6824c88.png) |

When the slider is set to 80%, the battery charges less because more of the volume is being reserved to respond to congestion.

**Comment by Sanne512** [comment]
> Changing the congestion percentage should have no effect on the behaviour of the battery in times of network congestion.

I don't understand this: because I think it does the change the behaviour of the battery in times of network congestion. If you have a really big battery and more storage volume of the battery is available for congestion (slider set to a lower percentage), more kWh can be stored in the battery which would otherwise have caused overcapacity. Could you explain to me what goes wrong in my reasoning?

**Comment by antw** [comment]
> If you have a really big battery and more storage volume of the battery is available for congestion (slider set to a lower percentage), more kWh can be stored in the battery which would otherwise have caused overcapacity.

That's a fair point; you're completely right. This does actually happen in your LES on 22nd April:

| 30% Reserved | 70% Reserved |
| --- | --- |
| <img width="560" alt="screen shot 2016-06-30 at 09 21 10" src="https://cloud.githubusercontent.com/assets/4383/16480039/4be0caf2-3ea4-11e6-915c-acbbd2ef89c4.png"> | <img width="575" alt="screen shot 2016-06-30 at 09 20 30" src="https://cloud.githubusercontent.com/assets/4383/16480044/559deafc-3ea4-11e6-9644-8e11bf5730d6.png"> |

In this case, because more of the battery is reserved, it has more empty volume to prevent the excess production coming from "Huizen achter buurtbatterij".

**Comment by Sanne512** [comment]
Ah great! Then i understood it correctly. I'll close this issue for now, since i wasn't able to reproduce this error either. If i encounter it again i'll reopen this issue.
